### PR TITLE
CI: Pin setuptools for python dev

### PR DESCRIPTION
**Pull request by phofl**
Lets see if this works

- [x] xref #44980

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
should this

https://github.com/pandas-dev/pandas/blob/d228a781b4d7be6c753cee940ce3ee692e97697d/ci/setup_env.sh#L109

also be pinned?

**Comment by phofl** [comment]
Yep, thx

**Comment by mroeschke** [comment]
Could you leave a `TODO` comment nearby the change referencing https://github.com/pypa/setuptools/issues/2941?

**Comment by phofl** [comment]
Done

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
merge when ready

**Comment by jbrockmendel** [comment]
uhh the ArrayManager tests have been going for 4 hours and counting

File: .github/workflows/python-dev.yml [hunk 1]
```diff
@@ -45,10 +45,11 @@ jobs:
       with:
         python-version: '3.10-dev'
 
+    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
```
Comment (jonashaag) [inline .github/workflows/python-dev.yml:48]:
```suggestion
    # See GH#44980 https://github.com/pypa/setuptools/issues/2941
```

File: .github/workflows/python-dev.yml [hunk 2]
```diff
@@ -45,10 +45,11 @@ jobs:
...
     - name: Install dependencies
       shell: bash
       run: |
-        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install --upgrade pip "setuptools<60.0.0" wheel
```
Comment (jonashaag) [inline .github/workflows/python-dev.yml:52]:
```suggestion
        python -m pip install --upgrade pip "setuptools<60.0.0|>=60.0.3" wheel
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -36,13 +36,14 @@ jobs:
     vmImage: ubuntu-18.04
 
   steps:
+    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
```
Comment (jonashaag) [inline azure-pipelines.yml:39]:
```suggestion
    # See GH#44980 https://github.com/pypa/setuptools/issues/2941
```

File: azure-pipelines.yml [hunk 4]
```diff
@@ -36,13 +36,14 @@ jobs:
...
     - script: |
         docker pull quay.io/pypa/manylinux2014_i686
         docker run -v $(pwd):/pandas quay.io/pypa/manylinux2014_i686 \
         /bin/bash -xc "cd pandas && \
         /opt/python/cp38-cp38/bin/python -m venv ~/virtualenvs/pandas-dev && \
         . ~/virtualenvs/pandas-dev/bin/activate && \
-        python -m pip install --no-deps -U pip wheel setuptools && \
+        python -m pip install --no-deps -U pip wheel 'setuptools<60.0.0' && \
```
Comment (jonashaag) [inline azure-pipelines.yml:46]:
```suggestion
        python -m pip install --no-deps -U pip wheel 'setuptools<60.0.0|>=60.0.3' && \
```

File: ci/setup_env.sh [hunk 5]
```diff
@@ -106,7 +106,8 @@ echo "[Build extensions]"
 python setup.py build_ext -q -j2
 
 echo "[Updating pip]"
-python -m pip install --no-deps -U pip wheel setuptools
+# TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
```
Comment (jonashaag) [inline ci/setup_env.sh:109]:
```suggestion
# See GH#44980 https://github.com/pypa/setuptools/issues/2941
```

File: ci/setup_env.sh [hunk 6]
```diff
@@ -106,7 +106,8 @@ echo "[Build extensions]"
...
+python -m pip install --no-deps -U pip wheel "setuptools<60.0.0"
```
Comment (jonashaag) [inline ci/setup_env.sh:110]:
```suggestion
python -m pip install --no-deps -U pip wheel "setuptools<60.0.0|>=60.0.3"
```

**Comment by alimcmaster1** [comment]
> uhh the ArrayManager tests have been going for 4 hours and counting

hmm 3hrs on second attempt.. 

Should be allow >=60.0.3 given this release fixes the issue?

**Review by jonashaag — COMMENTED** [review]
Fixed with 60.0.3

**Comment by phofl** [comment]
I think we should be able to avoid pinning this, if the newest version works?

**Comment by MarcoGorelli** [comment]
yes, agreed

**Comment by phofl** [comment]
Looks like it does not adress our problem, https://github.com/pandas-dev/pandas/runs/4594019360?check_suite_focus=true

File: .github/workflows/python-dev.yml (hunk 1 above)
Comment (phofl) [reply .github/workflows/python-dev.yml:48]: No we do not want to keep that

**Comment by phofl** [comment]
The experimental data manager issue seems unrelated. I can also see this at https://github.com/pandas-dev/pandas/runs/4594019873?check_suite_focus=true. This has the newest setuptools

**Comment by phofl** [comment]
experimental data manager failure is unrelated, see #44994

**Comment by jbrockmendel** [comment]
other than ArrayManager is this ready?

**Comment by phofl** [comment]
Yes

**Comment by jreback** [comment]
> Yes

ok and the AM tests broke on the json changes?

**Comment by jreback** [comment]
actually can we just unpin this? e.g. if >=60.0.3 its working?

**Comment by phofl** [comment]
60.0.3 does not work unfortunately. There are still problems. Posted some more info on the issue.

**Comment by phofl** [comment]
Yes AM case was from the json change

**Comment by jreback** [comment]
cool thanks @phofl 

ok if you don't mind disabling the failing tests in AM build for the json (and create an issue about it) to get the CI back to green :->

**Comment by phofl** [comment]
#44999

**Pull request merged**